### DISCUSSION TITLE
conflicting dependencies pleas help me ım noob

### DISCUSSION THREAD

**Discussion by mirket-bey** [BlackArch/blackarch#4343] [Support] [answered]
Type: Why can't I update

## Question

I'm using arch for the first time, I installed it on my disk and looked at how to update it, but what do I see, conflicting java. How do I fix it?

[ blackarch /home/liveuser/Desktop ]#    sudo pacman -Syu
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 blackarch is up to date
:: Starting full system upgrade...
:: Replace appstream-qt with extra/appstream-qt5? [Y/n] 
:: Replace dbus-python with extra/python-dbus? [Y/n] 
:: Replace kactivities with extra/kactivities5? [Y/n] 
:: Replace kactivities with extra/plasma-activities? [Y/n] 
:: Replace karchive with extra/karchive5? [Y/n] 
:: Replace kauth with extra/kauth5? [Y/n] 
:: Replace kbookmarks with extra/kbookmarks5? [Y/n] 
:: Replace kcodecs with extra/kcodecs5? [Y/n] 
:: Replace kcompletion with extra/kcompletion5? [Y/n] 
:: Replace kconfig with extra/kconfig5? [Y/n] 
:: Replace kconfigwidgets with extra/kconfigwidgets5? [Y/n] 
:: Replace kcoreaddons with extra/kcoreaddons5? [Y/n] 
:: Replace kcrash with extra/kcrash5? [Y/n] 
:: Replace kdbusaddons with extra/kdbusaddons5? [Y/n] 
:: Replace kdeclarative with extra/kdeclarative5? [Y/n] 
:: Replace kded with extra/kded5? [Y/n] 
:: Replace kglobalaccel with extra/kglobalaccel5? [Y/n] 
:: Replace kguiaddons with extra/kguiaddons5? [Y/n] 
:: Replace ki18n with extra/ki18n5? [Y/n] 
:: Replace kiconthemes with extra/kiconthemes5? [Y/n] 
:: Replace kio with extra/kio5? [Y/n] 
:: Replace kitemviews with extra/kitemviews5? [Y/n] 
:: Replace kjobwidgets with extra/kjobwidgets5? [Y/n] 
:: Replace knotifications with extra/knotifications5? [Y/n] 
:: Replace kpackage with extra/kpackage5? [Y/n] 
:: Replace kservice with extra/kservice5? [Y/n] 
:: Replace ktextwidgets with extra/ktextwidgets5? [Y/n] 
:: Replace kwallet with extra/kwallet5? [Y/n] 
:: Replace kwayland with extra/kwayland5? [Y/n] 
:: Replace kwidgetsaddons with extra/kwidgetsaddons5? [Y/n] 
:: Replace kwindowsystem with extra/kwindowsystem5? [Y/n] 
:: Replace kxmlgui with extra/kxmlgui5? [Y/n] 
:: Replace mlocate with extra/plocate? [Y/n] 
:: Replace plasma-framework with extra/libplasma? [Y/n] 
:: Replace plasma-framework with extra/plasma-framework5? [Y/n] 

:: Replace ruby-abbrev with extra/ruby? [Y/n] 
:: Replace ruby-base64 with extra/ruby? [Y/n] 
:: Replace ruby-benchmark with extra/ruby? [Y/n] 
:: Replace ruby-bigdecimal with extra/ruby? [Y/n] 
:: Replace ruby-bundledgems with extra/ruby-bundled-gems? [Y/n] 
:: Replace ruby-cgi with extra/ruby? [Y/n] 
:: Replace ruby-csv with extra/ruby? [Y/n] 
:: Replace ruby-date with extra/ruby? [Y/n] 
:: Replace ruby-delegate with extra/ruby? [Y/n] 
:: Replace ruby-did_you_mean with extra/ruby? [Y/n] 
:: Replace ruby-digest with extra/ruby? [Y/n] 
:: Replace ruby-drb with extra/ruby? [Y/n] 
:: Replace ruby-english with extra/ruby? [Y/n] 
:: Replace ruby-etc with extra/ruby? [Y/n] 
:: Replace ruby-fcntl with extra/ruby? [Y/n] 
:: Replace ruby-fiddle with extra/ruby? [Y/n] 
:: Replace ruby-fileutils with extra/ruby? [Y/n] 
:: Replace ruby-find with extra/ruby? [Y/n] 
:: Replace ruby-forwardable with extra/ruby? [Y/n] 
:: Replace ruby-getoptlong with extra/ruby? [Y/n] 
:: Replace ruby-io-console with extra/ruby? [Y/n] 
:: Replace ruby-io-nonblock with extra/ruby? [Y/n] 
:: Replace ruby-io-wait with extra/ruby? [Y/n] 
:: Replace ruby-ipaddr with extra/ruby? [Y/n] 
:: Replace ruby-json with extra/ruby? [Y/n] 
:: Replace ruby-logger with extra/ruby? [Y/n] 
:: Replace ruby-mutex_m with extra/ruby? [Y/n] 
:: Replace ruby-net-http with extra/ruby? [Y/n] 
:: Replace ruby-open-uri with extra/ruby? [Y/n] 
:: Replace ruby-psych with extra/ruby? [Y/n] 
:: Replace ruby-reline with extra/ruby? [Y/n] 
:: Replace ruby-ruby2_keywords with extra/ruby? [Y/n] 
:: Replace ruby-stringio with extra/ruby? [Y/n] 
:: Replace ruby-time with extra/ruby? [Y/n] 
:: Replace ruby-tmpdir with extra/ruby? [Y/n] 
:: Replace ruby-uri with extra/ruby? [Y/n] 
:: Replace solid with extra/solid5? [Y/n] 
:: Replace sonnet with extra/sonnet5? [Y/n] 
:: Replace tracker3 with extra/tinysparql? [Y/n] 
resolving dependencies...

:: There are 2 providers available for qt6-multimedia-backend:
:: Repository extra
   1) qt6-multimedia-ffmpeg  2) qt6-multimedia-gstreamer

Enter a number (default=1): ^[[F
error: invalid number: 
Enter a number (default=1): 1
looking for conflicting packages...
error: unresolvable package conflicts detected
error: failed to prepare transaction (conflicting dependencies)
:: jdk-openjdk and jre-openjdk are in conflict

my logs :3

**Comment by D3vil0p3r** [comment]
Are you sure you installed the system on disk? I see `liveuser` as user account...

**Comment by mirket-bey** [comment]
> Are you sure you installed the system on disk? I see `liveuser` as user account...

yes because I'm getting the same problem on disk and I came from debian and crash the system trying to fix it

**Comment by mirket-bey** [comment]
> > Are you sure you installed the system on disk? I see `liveuser` as user account...
> 
> yes because I'm getting the same problem on disk and I came from debian and fucked up the system trying to fix it

but the problem is still the same

**Comment by D3vil0p3r** [comment]
Which BlackArch ISO did you use to install the system? Full, slim, netinstall?

**Comment by mirket-bey** [comment]
> Which BlackArch ISO did you use to install the system? Full, slim, netinstall?

slim

**Comment by mirket-bey** [comment]
> Which BlackArch ISO did you use to install the system? Full, slim, netinstall?

I wonder if it's a bug.

**Comment by D3vil0p3r** [comment]
Your issue comes from some java dependencies of some pre-installed tools using jre and jdk that conflict each other. The next ISO files need to be refactored by considering these errors.

As quick workaround, open terminal and run:
```
sudo pacman -Rns calamares
sudo pacman -Rd --nodeps jre-openjdk jre-openjdk-headless
sudo pacman -Rns theharvester
sudo pacman -Sy archlinux-keyring
sudo pacman -Syyu
```
When it will ask you:
```
warning: removing 'jre11-openjdk' from target list because it conflicts with 'jdk11-openjdk'
warning: removing 'jre11-openjdk-headless' from target list because it conflicts with 'jdk11-openjdk'
:: jdk11-openjdk and jre11-openjdk are in conflict. Remove jre11-openjdk? [y/N]
:: jdk11-openjdk and jre11-openjdk-headless are in conflict. Remove jre11-openjdk-headless? [y/N]
```
press `y` in both of cases.

It should start to update packages. Finally, reinstall `theharvester` by:
```
sudo pacman -S theharvester
```
If it works, please close the issue.

**Comment by mirket-bey** [comment, marked as answer]
> Your issue comes from some java dependencies of some pre-installed tools using jre and jdk that conflict each other. The next ISO files need to be refactored by considering these errors.
> 
> As quick workaround, open terminal and run:
> 
> ```
> sudo pacman -Rns calamares
> sudo pacman -Rd --nodeps jre-openjdk jre-openjdk-headless
> sudo pacman -Rns theharvester
> sudo pacman -Sy archlinux-keyring
> sudo pacman -Syyu
> ```
> 
> When it will ask you:
> 
> ```
> warning: removing 'jre11-openjdk' from target list because it conflicts with 'jdk11-openjdk'
> warning: removing 'jre11-openjdk-headless' from target list because it conflicts with 'jdk11-openjdk'
> :: jdk11-openjdk and jre11-openjdk are in conflict. Remove jre11-openjdk? [y/N]
> :: jdk11-openjdk and jre11-openjdk-headless are in conflict. Remove jre11-openjdk-headless? [y/N]
> ```
> 
> press `y` in both of cases.
> 
> It should start to update packages. Finally, reinstall `theharvester` by:
> 
> ```
> sudo pacman -S theharvester
> ```
> 
> If it works, please close the issue.

thank you very much I am trying now

**Reply by mirket-bey** [reply]
thats work for me.

**Comment by mirket-bey** [comment]
[liveuser@blackarch]-[~/Desktop]
>>> sudo pacman -Rns calamares
checking dependencies...
error: failed to prepare transaction (could not satisfy dependencies)
:: removing calamares breaks dependency 'calamares' required by blackarch-config-calamares


> sudo pacman -Syyu

error: failed to prepare transaction (could not satisfy dependencies)
:: removing plasma-framework breaks dependency 'plasma-framework' required by calamares
:: removing appstream-qt breaks dependency 'appstream-qt' required by calamares

and yes, error again

**Comment by D3vil0p3r** [comment]
You are acting on the ISO Live Environment instead of the installed system. You must first install the system on your disk. When you are inside the Live Env, on your Desktop you should see an icon for Installation. Double click there, the installer goes up, follow the process, and after the installation, reboot, land on the INSTALLED system and run the commands I reported above.

**Reply by mirket-bey** [reply]
jesus no give another error now ım install disk 

 liblouis-3.31.0-1-x86_64                                                                                             2,4 MiB  2022 KiB/s 00:01 [########################################################################################] 100%
 Total (  48/1340)                                                                                                  537,6 MiB  5,99 MiB/s 01:30 [########################################################################################] 100%
error: failed retrieving file 'liblouis-3.31.0-1-x86_64.pkg.tar.zst.sig' from mirror.rackspace.com : Failed to connect to mirror.rackspace.com port 80 after 10001 ms: Timeout was reached
warning: failed to retrieve some files
error: failed to commit transaction (download library error)
Errors occurred, no packages were upgraded.

**Reply by D3vil0p3r** [reply]
Please be clearer. it is not clear where
you're doing it. Is it on the installed system or it occurred during System Installation (by Calamares installer)?

Edit by sudo `/etc/pacman.d/mirrorlist` file and remove that mirror. Retry the upgrade again

**Comment by mirket-bey** [comment]
> You are acting on the ISO Live Environment instead of the installed system. You must first install the system on your disk. When you are inside the Live Env, on your Desktop you should see an icon for Installation. Double click there, the installer goes up, follow the process, and after the installation, reboot, land on the INSTALLED system and run the commands I reported above.

Okay man, I'll do as you say, I'll start the installation now.

**Comment by ikstream** [comment]
As you closed and reopened it, do you still have an issue, if yes, which?

**Reply by mirket-bey** [reply]
I'm sorry, I encountered the same error when installing blackarch on another computer and I came here to go back to the solution, I must have pressed the wrong button, thanks for your attention, and could you please add this problem and its solution to your FAQ?

**Reply by D3vil0p3r** [reply]
Let @anunna to decide if adding the topic to BA Wiki FAQ
